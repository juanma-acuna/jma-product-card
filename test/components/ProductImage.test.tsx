import React from 'react';
import render from 'react-test-renderer';
import { ProductImage, ProductCard } from '../../src/components';
import { product2 } from '../data/products';

describe('ProductImage', () => {
  test('debe mostrar el componente con la imagen default', () => {
    const wrapper = render.create(<ProductImage img="una_imagen.jpg" />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('debe mostrar el componente correctamente con la imagen del producto', () => {
    const wrapper = render.create(
      <ProductCard product={product2}>{() => <ProductImage />}</ProductCard>
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
