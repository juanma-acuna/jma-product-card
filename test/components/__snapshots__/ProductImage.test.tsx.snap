// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage debe mostrar el componente con la imagen default 1`] = `
<img
  alt="Product"
  className="productImg undefined"
  src="una_imagen.jpg"
/>
`;

exports[`ProductImage debe mostrar el componente correctamente con la imagen del producto 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product"
    className="productImg undefined"
    src="../../assets/images/coffee-mug.jpg"
  />
</div>
`;
